src/test/scala/SystemTests.scala

```diff
@@ -1,19 +1,33 @@
 import org.scalatest.funsuite.AnyFunSuite
-import util.Logger
+import util.{Logger, PerformanceTimer}
+
 
 import java.io.{BufferedWriter, File, FileWriter}
+import scala.collection.mutable
 import scala.io.Source
 import scala.sys.process.*
 
 /** Add more tests by simply adding them to the programs directory. Refer to the existing tests for the expected
   * directory structure and file-name patterns.
   */
 class SystemTests extends AnyFunSuite {
+
+  val testPath = "./src/test/"
   val correctPath = "./src/test/correct"
   val correctPrograms: Array[String] = getSubdirectories(correctPath)
   val incorrectPath = "./src/test/incorrect"
   val incorrectPrograms: Array[String] = getSubdirectories(incorrectPath)
 
+  case class TestResult(passed: Boolean, verified: Boolean, shouldVerify: Boolean, hasExpected: Boolean, timedOut: Boolean, matchesExpected: Boolean, translateTime: Long, verifyTime: Long) {
+    val toCsv = s"$passed,$verified,$shouldVerify,$hasExpected,$timedOut,$matchesExpected,$translateTime,$verifyTime"
+  }
+
+  object TestResult {
+    val csvHeader = "passed,verified,shouldVerify,hasExpected,timedOut,matchesExpected,translateTime,verifyTime"
+  }
+
+  val testResults: mutable.Map[String, TestResult] = mutable.HashMap()
```

Comment:
Is there a reason for using a Map here, given that it means the output isn't ordered, which is much less convenient? It doesn't seem like we care about the constant lookup time?

Reply:
Uses an ArrayBuffer now